### PR TITLE
hpilo_fw: parse: use correct bytes handling

### PR DESCRIPTION
**Pull request by buhman**

### CODE REVIEW

File: hpilo_fw.py [hunk 1]
```diff
@@ -62,10 +62,10 @@ def download(ilo, path=None, progress = lambda txt: None):
     return False
 
 def parse(fwfile, ilo):
-    fd = open(fwfile)
+    fd = open(fwfile, 'rb')
     data = fd.read()
     fd.close()
-    if '_SKIP=' in data:
+    if b'_SKIP=' in data:
```
Comment (buhman) [inline hpilo_fw.py]: This search should probably be bounded to a much smaller subset of `data`.

File: hpilo_fw.py [hunk 2]
```diff
@@ -62,10 +62,10 @@ def download(ilo, path=None, progress = lambda txt: None):
     return False
 
 def parse(fwfile, ilo):
-    fd = open(fwfile)
+    fd = open(fwfile, 'rb')
```
Comment (seveas) [inline hpilo_fw.py]: This is a good change

File: hpilo_fw.py (hunk 1 above)
Comment (seveas) [inline hpilo_fw.py]: This change unfortunately needs to be done in a different way (no b'' strings), as I still support python 2.4 for the poor sods stuck on rhel 5.

Comment (buhman) [inline hpilo_fw.py]: I'm not familiar with python 2.4--what's the correct way to specify forwards-compatible byte strings? We could do something like:

```python
if '_SKIP='.encode('utf-8') in data:
```

But that seems rather hacky.

**Comment by seveas** [comment]
Better version now pushed as a2f22b5, thanks for reporting!

**Pull request closed without merging**